daisy_workflows/image_build/install_package/cos/package_replacement/compile_debian_package.sh

```diff
@@ -0,0 +1,167 @@
+  git clone https://github.com/GoogleCloudPlatform/guest-agent.git --branch ${guest_agent_version}
+  mv ./board-overlays/project-lakitu/app-admin/google-guest-agent/files ./guest-agent
+  cd guest-agent
+
+  echo -e "\nATTENTION: Applying patches...\n"
+  search_dir=./files
+  for file in "$search_dir"/*
+  do
+    if [[ "$file" == *"patch"* ]]; then
+      git apply "$file"
+    fi
+  done
+}
+
+install_build_deps() {
+  # Install the build dependencies.
+  echo -e "\nATTENTION: Installing build deps...\n"
+  apt-get -y update
+  apt-get install -y --no-install-{suggests,recommends} git-core \
+    debhelper devscripts build-essential equivs libdistro-info-perl
+  export PKGNAME="google-guest-agent"
+  mk-build-deps -t "apt-get -o Debug::pkgProblemResolver=yes \
+    --no-install-recommends --yes" --install packaging/debian/control
+  dpkg-checkbuilddeps packaging/debian/control
+}
+
+find_debian_version() {
+  # Identify and store the debian version.
+  echo -e "\nATTENTION: Finding the debian version...\n"
+  cat /etc/debian_version
+  DEB_VERSION=$(</etc/debian_version)
+  # deb_version=${DEB}
+  if [[ "$DEB_VERSION" =~ "bookworm" ]]; then
+      export DEB="12"
+  else
+      export DEB="11"
+  fi
+}
+
+install_go(){
+  # Install the correct version of Go.
+  echo -e "\nATTENTION: Installing go...\n"
+  cd /workspace
+  source ./install_go.sh; install_go /tmp/go
+}
+
+upstream_tarball() {
+  # Create the build directory and the upstream tarball.
+  cd guest-agent
+  mkdir /tmp/debpackage
+  export BUILD_DIR="/tmp/debpackage"
+  export VERSION="112723"
+  export TAR="${PKGNAME}_${VERSION}.orig.tar.gz"
+  echo -e "\nATTENTION: Creating build dir and upstream tarball...\n"
+  tar czvf "${BUILD_DIR}/${TAR}" --exclude .git --exclude packaging \
+    --transform "s/^\./${PKGNAME}-${VERSION}/" .
+}
+
+build_tarball(){
+  # Extract and build the tarball.
+  echo -e "\nATTENTION: Extracting tarball and building...\n"
+  tar -C "$BUILD_DIR" -xzvf "${BUILD_DIR}/${TAR}"
+  cp -r packaging/debian "${BUILD_DIR}/${PKGNAME}-${VERSION}/"
+  cd "${BUILD_DIR}/${PKGNAME}-${VERSION}"
+  [[ -f debian/changelog ]] && rm debian/changelog
+  export RELEASE="g1${DEB}"
+  dch --create -M -v 1:${VERSION}-${RELEASE} --package $PKGNAME -D stable \
+    "Debian packaging for ${PKGNAME}"
+  debuild -us -uc
+  cd ..
+}
+
+unpack_binaries(){
+  # Unpack the binaries into a local directory that contains 1) the .deb files
+  # (which provide installation paths) and 2) binaries in a localized dir format.
+  echo -e "\nATTENTION: Unpacking binaries in a directory...\n"
+  dpkg-deb -x google-guest-agent_112723-g111_amd64.deb debian_binaries
+  dpkg-deb -c google-guest-agent_112723-g111_amd64.deb >> google-guest-agent-readable-deb.txt
+  echo -e "\nATTENTION: google-guest-agent.deb file contents below...\n"
+  cat google-guest-agent-readable-deb.txt
+  mv debian_binaries /workspace/upload
+  mv google-guest-agent-readable-deb.txt /workspace
+}
+
+identify_replacement_files(){
```

Comment:
are you identifying replacements of what, and where?

Reply:
Yes, the .deb file has a list of files to install for the guest agent. We only really want to replace the .service and binaries (not the licenses) for testing purposes, so we go through the list of files in .deb file and skip the licenses. This will tell us what files to install (like google-guest-agent.service) and where to install them (/usr/bin).